lib/useQuery.js

```diff
@@ -110,15 +116,19 @@ export function useQuery(serviceName, options = {}, queryHookOptions = {}) {
       transformResponse,
     })
       .then(res => {
-        if (reqRef === requestRef.current) {
+        if (state.dirty) {
```

Comment:
It is not correct / valid to read state.dirty in a promise callback because this state variable is a stale closure - this will not reflect the state as of now, but as it was when the effect was first triggered.
So while the tests seem to be passing, I don't think this is 100% correct. 🤔 But also - really don't know how to correct it - the new branch where state is managed outside React is much easier to reason about.. should we get that over the finish line??